test/gleam/regex_test.gleam

```diff
@@ -44,3 +45,32 @@ pub fn match_test() {
   regex.match(re, "boo")
   |> should.equal(False)
 }
+
+pub fn split_test() {
+  assert Ok(re) = regex.from_string(" *, *")
+
+  regex.split(re, "foo,32, 4, 9  ,0")
+  |> should.equal(["foo", "32", "4", "9", "0"])
+}
+
+pub fn scan_test() {
+  assert Ok(re) = regex.from_string("[oi]n a(.?) (\\w+)")
+
+  regex.scan(re, "I am on a boat in a lake.")
+  |> should.equal(
+    [
+      Match(
+        match: "on a boat",
+        index: 5,
+        number: 1,
+        submatches: [None, Some("boat")],
+      ),
+      Match(
+        match: "in a lake",
+        index: 15,
+        number: 2,
```

Comment:
Is this number always simply the position in the list

Reply:
https://github.com/gleam-lang/stdlib/pull/83#discussion_r444722213